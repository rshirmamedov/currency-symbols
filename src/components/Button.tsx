import { ButtonHTMLAttributes } from "react";
import { Loading } from "./Loading";

type ButtonProps = {
    title: string;
} & ButtonHTMLAttributes<HTMLButtonElement>;

export const Button = ({ title, ...buttonProps }: ButtonProps) => {
    return (
        <button
            className="w-full py-2.5 px-4 text-sm rounded text-white bg-blue-600 hover:bg-blue-700 disabled:bg-blue-400 focus:outline-none mt-6"
            {...buttonProps}
        >
            {title}
        </button>
    );
};
