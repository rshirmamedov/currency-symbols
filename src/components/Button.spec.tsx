import { render } from "@testing-library/react";
import { Button } from "./Button";

describe("Button", () => {
    it("renders", () => {
        const { container } = render(<Button title="Login" />);

        expect(container).toMatchSnapshot();
    });
});