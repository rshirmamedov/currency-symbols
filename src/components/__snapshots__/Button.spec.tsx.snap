// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<div>
  <button
    class="w-full py-2.5 px-4 text-sm rounded text-white bg-blue-600 hover:bg-blue-700 disabled:bg-blue-400 focus:outline-none mt-6"
  >
    Login
  </button>
</div>
`;
